src/Workspaces/Remote/ServiceHub/Services/CodeAnalysisService_AddImport.cs

```diff
@@ -61,26 +61,26 @@ public SymbolSearchService(CodeAnalysisService codeAnalysisService)
             public async Task<IList<PackageWithTypeResult>> FindPackagesWithTypeAsync(
                 string source, string name, int arity, CancellationToken cancellationToken)
             {
-                var result = await codeAnalysisService.Rpc.InvokeAsync<IList<PackageWithTypeResult>>(
-                    nameof(FindPackagesWithTypeAsync), source, name, arity).ConfigureAwait(false);
+                var result = await codeAnalysisService.InvokeAsync<IList<PackageWithTypeResult>>(
+                    nameof(FindPackagesWithTypeAsync), new object[] { source, name, arity }, cancellationToken).ConfigureAwait(false);
```

Comment:
but you added cancellationToken here...

Reply:
because the cancellation token was given, but existing code (FindPackagesWithTypeAsync) somehow didn't use it. do you want me to remove the cancellation token? since it was there, I thought it was just a bug where it miss using it. but if it was omitted on purpose, I can remove using cancellation.

ublic async Task<IList<PackageWithTypeResult>> FindPackagesWithTypeAsync(
                string source, string name, int arity, CancellationToken cancellationToken)	                string source, string name, int arity, CancellationToken cancellationToken)
            {	            {
                var result = await codeAnalysisService.Rpc.InvokeAsync<IList<PackageWithTypeResult>>(	                var result = await codeAnalysisService.InvokeAsync<IList<PackageWithTypeResult>>(
                    nameof(FindPackagesWithTypeAsync), source, name, arity).ConfigureAwait(false);	                    nameof(FindPackagesWithTypeAsync), new object[] { source, name, arity }, cancellationToken).ConfigureAwait(false);

Reply:
>because the cancellation token was given,

Right, my point is, this is what we should be doing in all cases.  

All of tehse are async codepaths taht are cancellation safe and would want an RPC invocation to cancel as quickly as possible if requested.

Reply:
Note: this is a good example of why i would never make cancellation tokens optional in our existing APIs.  It's too easy to miss them.  Especially as things changed here, we never ensured that all the clients of these APIs were appropriately being explicit about the behavior they wanted.  Would it be possible to deprecate (with a warning) the existing methods that don't take cancellation tokens?  it's likely to always be a bug to call that one.